### PR TITLE
x86_cpu_flags: update case's step

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1953412

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap.q35: PASS (78.62 s)

**Comment by nanliu-r** [comment]
@vivianQizhu @huangyum Please help to review this, thanks a lot.

**Comment by huangyum** [comment]
@nanliu-r Would you pls update as we discussed in bugzilla?

**Comment by nanliu-r** [comment]
> @nanliu-r Would you pls update as we discussed in bugzilla?

Sorry for late update, I will update this tomorrow, thanks.

**Comment by nanliu-r** [comment]
Test result like this:
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: CANCEL: 'Opteron_G3' doesn't support this test case (2.96 s)

**Comment by nanliu-r** [comment]
@vivianQizhu @huangyum Would you please help to review this again? thanks a lot.

File: qemu/tests/x86_cpu_flags.py [hunk 1]
```diff
@@ -22,6 +22,13 @@ def run(test, params, env):
     if check_host_flags:
         check_cpu_flags(params, flags, test)
 
+    no_models = params.get("no_models", "")
+    cpu_model = params['cpu_model']
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flags.py:26]: Not all flag test cases define `cpu_model`, right? Then they might hit error here.

File: qemu/tests/cfg/x86_cpu_flags.cfg [hunk 2]
```diff
@@ -94,6 +94,7 @@
                 - test_smap:
                     # support RHEL.7 guest or later
                     no RHEL.6
+                    no_models = ["Haswell", "Haswell-noTSX", "IvyBridge", "SandyBridge", "Westmere", "Nehalem", "Penryn", "Conroe", "Opteron_G4", "Opteron_G3", "Opteron_G2", "Opteron_G1"]
```
Comment (huangyum) [inline qemu/tests/cfg/x86_cpu_flags.cfg:97]: Prefer 'unsupported_models'.

File: qemu/tests/x86_cpu_flags.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flags.py]: > Not all flag test cases define `cpu_model`, right? Then they might hit error here.

Thanks and update, run one other case random and can PASS.

File: qemu/tests/cfg/x86_cpu_flags.cfg (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/cfg/x86_cpu_flags.cfg]: > Prefer 'unsupported_models'.

updated and thanks.

File: qemu/tests/cfg/x86_cpu_flags.cfg [hunk 3]
```diff
@@ -94,6 +94,7 @@
                 - test_smap:
                     # support RHEL.7 guest or later
                     no RHEL.6
+                    unsupported_models = ["Haswell", "Haswell-noTSX", "IvyBridge", "SandyBridge", "Westmere", "Nehalem", "Penryn", "Conroe", "Opteron_G4", "Opteron_G3", "Opteron_G2", "Opteron_G1"]
```
Comment (huangyum) [inline qemu/tests/cfg/x86_cpu_flags.cfg:97]: unsupported_models = "Haswell Haswell-noTSX ..."

File: qemu/tests/x86_cpu_flags.py [hunk 4]
```diff
@@ -22,6 +22,13 @@ def run(test, params, env):
     if check_host_flags:
         check_cpu_flags(params, flags, test)
 
+    unsupported_models = params.get("unsupported_models", "")
+    cpu_model = params.get("cpu_model")
+    if not cpu_model:
+        cpu_model = cpu.get_qemu_best_cpu_model(params)
+    if cpu_model in unsupported_models:
```
Comment (huangyum) [inline qemu/tests/x86_cpu_flags.py:29]: `if cpu_model in unsupported_models.split()`

File: qemu/tests/cfg/x86_cpu_flags.cfg (hunk 3 above)
Comment (nanliu-r) [reply qemu/tests/cfg/x86_cpu_flags.cfg]: update and thanks.

File: qemu/tests/x86_cpu_flags.py (hunk 4 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flags.py]: update and thanks.
test PASS: (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: PASS (71.39 s)

Comment (huangyum) [reply qemu/tests/x86_cpu_flags.py]: Pls update polarion test case as well, so they can be consistent.

Comment (nanliu-r) [reply qemu/tests/x86_cpu_flags.py]: > Pls update polarion test case as well, so they can be consistent.

update and thanks.

File: qemu/tests/cfg/x86_cpu_flags.cfg [hunk 5]
```diff
@@ -94,6 +94,7 @@
                 - test_smap:
                     # support RHEL.7 guest or later
                     no RHEL.6
+                    unsupported_models = "Haswell Haswell-noTSX IvyBridge SandyBridge Westmere Nehalem Penryn Conroe Opteron_G4 Opteron_G3 Opteron_G2 Opteron_G1"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/x86_cpu_flags.cfg:97]: @nanliu-r Shall we split it into two lines?

Comment (nanliu-r) [reply qemu/tests/cfg/x86_cpu_flags.cfg]: > @nanliu-r Shall we split it into two lines?

@vivianQizhu  I split AMD and Intel into different lines, thanks.

**Review by huangyum — APPROVED** [review]
ACK

**Pull request merged**